### PR TITLE
Update breadth_first_search.py

### PR DESCRIPTION
**Pull request by yashvarshney003**
Issue id #364 
AttributeError: 'Node' object has no attribute 'parent_index'
Fixed !!!.

### CODE REVIEW

**Comment by wuxiaohua1011** [comment]
Also for DFS

**Comment by AtsushiSakai** [comment]
Hi. Thank you for fixing it. I have two requests before merge.
1. Could you please change all `pind` to `parent_index`, I think that is more intuitive.
2. Could you please add two test files for both code? Maybe this code will help you:
https://github.com/AtsushiSakai/PythonRobotics/blob/master/tests/test_rrt_dubins.py

**Comment by yashvarshney003** [comment]
I added two test file  test_breadth_first_search.py and test_depth_first_search.py in test folder

File: tests/test_rrt_dubins.py [hunk 1]
```diff
@@ -3,8 +3,8 @@
 import os
 sys.path.append(os.path.dirname(os.path.abspath(__file__)) +
                 "/../PathPlanning/RRTDubins/")
-sys.path.append(os.path.dirname(os.path.abspath(__file__)) +
-                "/../PathPlanning/DubinsPath/")
+#sys.path.append(os.path.dirname(os.path.abspath(__file__)) +
```
Comment (AtsushiSakai) [inline tests/test_rrt_dubins.py:6]: Why did you change it?

File: tests/test_dijkstra.py [hunk 2]
```diff
@@ -1,12 +1,19 @@
...
+    raise
 
 print(__file__)
 
 
 class Test(TestCase):
 
     def test1(self):
-        m.show_animation = False
+        m.show_animation = True
```
Comment (AtsushiSakai) [inline tests/test_dijkstra.py:18]: This should be False. I thinkg

File: tests/test_breadth_first_search.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+from unittest import TestCase
+import sys
+import os
+sys.path.append(os.path.dirname(os.path.abspath(__file__)) +
+                "/../PathPlanning/BreadthFirstSearch/")
+
+
+try:
+    import breadth_first_search as m
+except:
```
Comment (AtsushiSakai) [inline tests/test_breadth_first_search.py:10]: This should be "except ImportError" to fix CI error.

File: tests/test_breadth_first_search.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+class Test(TestCase):
+
+    def test1(self):
+        m.show_animation = False
+        m.main()
+
+
+if __name__ == '__main__':  # pragma: no cover
+    test = Test()
+    test.test1()
```
Comment (AtsushiSakai) [inline tests/test_breadth_first_search.py:26]: Empty line is needed to fix CI error

File: tests/test_depth_first_search.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+from unittest import TestCase
+import sys
+import os
+sys.path.append(os.path.dirname(os.path.abspath(__file__)) +
+                "/../PathPlanning/DepthFirstSearch/")
+
+
+try:
+    import depth_first_search as m
+except:
```
Comment (AtsushiSakai) [inline tests/test_depth_first_search.py:10]: This should be "except ImportError" to fix CI error.

File: tests/test_dijkstra.py [hunk 6]
```diff
@@ -1,12 +1,19 @@
 from unittest import TestCase
 
-from PathPlanning.Dijkstra import dijkstra as m
+import sys
+import os
+sys.path.append(os.path.dirname(os.path.abspath(__file__)) +
+                "/../PathPlanning/Dijkstra/")
+try:
+    import dijkstra as m
+except:
```
Comment (AtsushiSakai) [inline tests/test_dijkstra.py:9]: This should be "except ImportError" to fix CI error.

File: tests/test_breadth_first_search.py (hunk 4 above)
Comment (yashvarshney003) [reply tests/test_breadth_first_search.py:26]: I don't understand.

**Comment by AtsushiSakai** [comment]
Could you please fix the CI errors?

File: tests/test_rrt_dubins.py (hunk 1 above)
Comment (yashvarshney003) [reply tests/test_rrt_dubins.py]: Programm is working fine without this line.

**Comment by yashvarshney003** [comment]
Now the CI errors are clear. But what is this? Help !!

**Comment by AtsushiSakai** [comment]
You don't care about it, this is test coverage ratio was decreased  because you added some new test codes.
I will merge it. Thank you!!
